lib/python/picongpu/plugins/data/compare.py

```diff
@@ -0,0 +1,222 @@
+                                       range_dict[key]['values'][0], 2)
+            try:
+                if observable is not "emittance":
+                    counts, bins = EnergyHistogramData(sim).get(
+                                   species, species_filter, iteration)
+                    start = 80
+                    limitE = 500
+                    data = counts[start:]
+                    for i in np.arange(data.size):
+                        if (max(data) == data[i]):
+                            peakEnergy = i
+                    for j in np.arange(data.size-peakEnergy):
+                        if (max(data)/3 - data[peakEnergy + j] > 0.0):
+                            FWHM_max = ((peakEnergy + j + start) *
+                                        limitE / len(bins))
+                            break
+                    for k in np.arange(peakEnergy):
+                        if (max(data)/3-data[peakEnergy - k] > 0.0):
+                            FWHM_min = ((peakEnergy - k + start)
+                                        * limitE / len(bins))
+                            break
+                if observable == "bunchEnergy":
+                    maxEnergy = 0.5 * (FWHM_max + FWHM_min)
+                    value = maxEnergy  # unit: MeV
+                if observable == "energySpread":
+                    energySpread = (FWHM_max - FWHM_min)
+                    value = energySpread  # unit: MeV
+                if observable == "energySpread/bunchEnergy":
+                    energySpread = (FWHM_max - FWHM_min)
+                    maxEnergy = 0.5 * (FWHM_max + FWHM_min)
+                    value = energySpread/maxEnergy
+                if observable == "maxEnergy":
+                    csum = np.cumsum(data[::-1])
+                    # 6.25e6 = 1pC
+                    maxE = ((start + np.shape(
+                            data[[csum > 6.25e6][::-1]])[0])
+                            * limitE / len(bins))
+                    value = maxE   # unit: MeV
+                if observable == "charge":
+                    for j in np.arange(data.size-peakEnergy):
+                        if (max(data) / 3 - data[peakEnergy + j] > 0.0):
+                            cmax = (peakEnergy + j)
+                            break
+                    for k in np.arange(peakEnergy):
+                        if (max(data) / 3 - data[peakEnergy - k] > 0.0):
+                            cmin = (peakEnergy - k)
+                            break
+                    # charge within +/- 1/3 peakEnergy, unit pC
+                    charge = data[cmin:cmax].sum() * 1.6021766209e-19 * 1.e12
+                    value = charge
+                if observable == "emittance":
+                    counts, bins, ts = EmittanceData(sim).get(
+                                       species, species_filter)
+                    emit = counts[iteration[0]][0] * 1.e6
+                    value = emit  # unit: pi mm mrad
+            except:
```

Comment:
That's quite a large try-except block. Where do you expect something to throw? In the data constructors or the `.get()`ters?

We know what we usually throw there, so should do something more precise like:
```python
except (ValueError, IOError, IndexError):
```

or you could also handle them individually to throw a more precise error to the user than "Error in sim" :-)